kernel/src/process_standard.rs

```diff
@@ -628,11 +628,14 @@ impl<C: Chip> Process for ProcessStandard<'_, C> {
         {
             // TODO: Check for buffer aliasing here
 
-            // Valid buffer, we need to adjust the app's watermark
-            // note: in_app_owned_memory ensures this offset does not wrap
-            let buf_end_addr = buf_start_addr.wrapping_add(size);
-            let new_water_mark = cmp::max(self.allow_high_water_mark.get(), buf_end_addr);
-            self.allow_high_water_mark.set(new_water_mark);
+            if self.in_app_owned_memory(buf_start_addr, size) {
+                // Valid buffer, if this is a memory buffer,
```

Comment:
Great find! I'm not sure I would've understood "memory buffer" at first glance though. I'd propose to clarify that we are not talking about a read-only memory section given flash is also memory.

```suggestion
                // Valid buffer, if this is not in a read-only (e.g. flash) memory section,
```